#!/usr/bin/env python
PACKAGE = "rm_shooter_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("block_effort", double_t, 0, "Trigger block effort", 0.95, 0.0, 10)
gen.add("block_duration", double_t, 0, "Trigger block duration", 0.05, 0.0, 2.0)
gen.add("block_speed", double_t, 0, "Trigger block speed", 0.5, 0.0, 5.)
gen.add("block_overtime", double_t, 0, "Trigger block overtime", 0.5, 0.0, 5.)
gen.add("anti_block_angle", double_t, 0, "Trigger anti block angle", 0.35, 0.0, 1.5)
gen.add("anti_block_threshold", double_t, 0, "Trigger anti block error threshold", 0.05, 0.0, 0.2)
gen.add("forward_push_threshold",double_t,0,"The trigger position threshold to push forward in push mode",0.01,0.0,1)
gen.add("exit_push_threshold",double_t,0,"The trigger position threshold to exit push mode",0.02,0.0,1)
gen.add("qd_10", double_t, 0, "Friction wheel rotation speed when bullet speed is 10m/s", 300, 0.0, 9999)
gen.add("qd_15", double_t, 0, "Friction wheel rotation speed when bullet speed is 15m/s", 300, 0.0, 9999)
gen.add("qd_16", double_t, 0, "Friction wheel rotation speed when bullet speed is 16m/s", 300, 0.0, 9999)
gen.add("qd_18", double_t, 0, "Friction wheel rotation speed when bullet speed is 18m/s", 300, 0.0, 9999)
gen.add("qd_30", double_t, 0, "Friction wheel rotation speed when bullet speed is 30m/s", 300, 0.0, 9999)
gen.add("lf_extra_rotat_speed", double_t, 0, "Lefrt friction extra rotation speed", -200, 0.0, 200)

exit(gen.generate(PACKAGE, "shooter", "Shooter"))
